feat: add graceful support for max Celo votes

andyhass opened the pull request:
<!--
Thank you for your contribution! 👍
Please make sure to read CONTRIBUTING.md if you have not already.
Disclaimer: Pull Requests that do not comply with the rules will be arbitrarily closed.
-->

### 📝 Description

There are a maximum number of Celo validator groups that can be voted for.  Previously, this error was not handled that well.  The user would be able to select the option to vote, but would then see this error message when choosing an amount to vote:

<img width="1134" alt="Screen Shot 2022-08-04 at 12 28 33 PM" src="https://user-images.githubusercontent.com/1444665/183112129-5bb5777f-280e-4573-aaf9-f2b73c65c883.png">

With this change, we indicate that the user cannot vote in staking management dashboard and prevent them from getting to the voting flow:

<img width="1317" alt="Screen Shot 2022-08-05 at 10 33 50 AM" src="https://user-images.githubusercontent.com/1444665/183112360-d5910a62-0411-4e6d-bfc8-5278b1796779.png">


### ❓ Context

- **Impacted projects**: `LLD, LLC` <!-- The list of end user projects impacted by the change. -->
- **Linked resource(s)**: `` <!-- Attach any ticket number if relevant. (JIRA / Github issue number) -->

### ✅ Checklist

- [x] **Test coverage** <!-- Are your changes covered by tests? Features must be tested, bugfixes must include a test that would have detected the issue. -->
- [x] **Atomic delivery** <!-- Is this pull request standalone? In order words, does it depend on nothing else? Please explain if not checked. -->
- [x] **No breaking changes** <!-- If there are breaking changes, please explain why. -->

### 📸 Demo

See above

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

<!-- If any of the expectations are not met please explain the reason in detail. -->

aelmilligy-ledger commented:
Hey @andyhass,
Thank you for your contribution. Some things to add:
- Could you add changeset (pnpm changeset)?
- Have a look on the conflicting files
Best regards.

andyhass commented:
> Hey @andyhass, Thank you for your contribution. Some things to add:
> 
> * Could you add changeset (pnpm changeset)?
> * Have a look on the conflicting files
>   Best regards.

Hey @aelmilligy-ledger - thanks for reviewing.  I've added the changelog and fixed up the merge conflicts as requested 😄

haammar-ledger commented on apps/ledger-live-desktop/src/renderer/families/celo/ManageModal/ManageModal.jsx:
```diff
@@ -45,9 +46,10 @@ const ManageModal = ({ name, account, ...rest }: Props) => {
     [dispatch, account],
   );
-
+  const groupsVotedFor = [...new Set(celoResources.votes.map(v => v.validatorGroup))];
```
Maybe `useMemo` here?

haammar-ledger replied:
Maybe `useMemo` would be useful here?...
